### PR TITLE
Create a chat list screen 

### PR DESCRIPTION
**Pull request by hodamohammadi**
includes changes from https://github.com/hodamohammadi/Conveyor/pull/14 , review that first please.

- ChatsListFragment to display the list of user's chats
- DefaultDialog model for a single dialog item 

Need to:
- update the models as the current models have two values for each info item
- add dialog items to the dialog adapter once firebase data is returned

### CODE REVIEW

File: conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * Fragment for a list of user's chats.
+ */
+class ChatsListFragment : Fragment(), DialogsListAdapter.OnDialogClickListener<IDialog<IMessage>>,
+        DialogsListAdapter.OnDialogLongClickListener<IDialog<IMessage>> {
+
+    private lateinit var dialogListAdapter: DialogsListAdapter<IDialog<IMessage>>
+    private lateinit var chatViewModel: ChatViewModel
+
+    override fun onCreateView(
+            inflater: LayoutInflater,
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt#L29"></span>Unexpected indentation (expected 8, actual 12)</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            container: ViewGroup?,
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt#L30"></span>Unexpected indentation (expected 8, actual 12)</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            savedInstanceState: Bundle?
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt#L31"></span>Unexpected indentation (expected 8, actual 12)</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        dialogListAdapter.setOnDialogClickListener(this)
+        dialogListAdapter.setOnDialogLongClickListener(this)
+
+        dialogsList.setAdapter(dialogListAdapter)
+    }
+
+    override fun onDialogClick(dialog: IDialog<IMessage>?) {
+        chatViewModel.threadId = dialog!!.id
+        // TODO: Launch SingleChatFragment
+
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/ChatsListFragment.kt#L56"></span>Unexpected blank line(s) before “}”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt [hunk 5]
```diff
@@ -21,7 +23,8 @@ class SingleChatFragment : Fragment(), MessageInput.InputListener, MessageInput.
         MessageInput.TypingListener {
 
     private lateinit var messagesAdapter: MessagesListAdapter<IMessage>
-
+    private lateinit var chatViewModel: ChatViewModel
+
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt#L27"></span>Trailing space(s)</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultDialog.kt [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package com.hodamohammadi.conveyor.models
+
+import com.stfalcon.chatkit.commons.models.IDialog
+import com.stfalcon.chatkit.commons.models.IMessage
+import com.stfalcon.chatkit.commons.models.IUser
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultDialog.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultDialog.kt#L5"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultDialog.kt [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+ * Default model for a Dialog object.
+ */
+class DefaultDialog(
+    val dialogImage: String = "",
+    val dialogUnreadCount: Int = 0,
+    val dialogId: String = "",
+    val dialogUsers: MutableList<DefaultUser> = mutableListOf(),
+    var dialogLastMessage: IMessage? = null,
+    val dialogNickname: String = ""
+): Serializable, IDialog<IMessage> {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultDialog.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultDialog.kt#L18"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt [hunk 8]
```diff
@@ -9,10 +9,10 @@ import java.util.Date
...
 data class DefaultMessage(
-    val messageId: String,
-    val messageText: String,
-    val messageCreatedAt: Date,
-    val messageUser: DefaultUser
+    val messageId: String = "",
+    val messageText: String = "",
+    val messageCreatedAt: Date = Date(),
+    val messageUser: DefaultUser = DefaultUser()
 ): Serializable, IMessage {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt#L16"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt [hunk 9]
```diff
@@ -1,15 +1,18 @@
 package com.hodamohammadi.conveyor.models
 
+import com.stfalcon.chatkit.commons.models.IDialog
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt#L3"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt [hunk 10]
```diff
@@ -1,15 +1,18 @@
...
+import com.stfalcon.chatkit.commons.models.IMessage
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt#L4"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt [hunk 11]
```diff
@@ -1,15 +1,18 @@
...
  */
 data class DefaultUser(
-    val userId: String,
-    val userName: String?,
-    val userAvatar: String?
+    val userId: String = "",
+    val userName: String? = "",
+    val userAvatar: String? = "",
+    val dialogs: List<DefaultDialog>? = emptyList()
 ): Serializable, IUser {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt#L16"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/utils/FirebaseHelper.kt [hunk 12]
```diff
@@ -8,9 +8,12 @@ import com.google.firebase.database.DataSnapshot
 import com.google.firebase.database.DatabaseReference
 import com.google.firebase.database.FirebaseDatabase
 import com.google.firebase.database.ValueEventListener
+import com.hodamohammadi.conveyor.models.DefaultDialog
 import com.hodamohammadi.conveyor.models.DefaultMessage
 import com.hodamohammadi.conveyor.models.DefaultUser
+import com.stfalcon.chatkit.commons.models.IDialog
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/utils/FirebaseHelper.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/utils/FirebaseHelper.kt#L14"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/utils/FirebaseHelper.kt [hunk 13]
```diff
@@ -8,9 +8,12 @@ import com.google.firebase.database.DataSnapshot
...
 import com.stfalcon.chatkit.commons.models.IMessage
+import com.stfalcon.chatkit.commons.models.IUser
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/utils/FirebaseHelper.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/utils/FirebaseHelper.kt#L16"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/utils/FirebaseHelper.kt [hunk 14]
```diff
@@ -56,8 +59,42 @@ class FirebaseHelper {
...
+//                            }
+//                        }
+//
+//                        override fun onCancelled(databaseError: DatabaseError) {
+//                        }
+//                    })
+
+            return threads
         }
+
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/utils/FirebaseHelper.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/utils/FirebaseHelper.kt#L98"></span>Unexpected blank line(s) before “}”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt [hunk 15]
```diff
@@ -0,0 +1,10 @@
+package com.hodamohammadi.conveyor.viewmodels
+
+import android.arch.lifecycle.ViewModel
+
+/**
+ * View model for communication between chat UIs.
+ */
+class ChatViewModel: ViewModel() {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/9a67d17b3b4351e9a1c31b5b5c4883ada984b461/conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt#L8"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt [hunk 16]
```diff
@@ -0,0 +1,17 @@
+package com.hodamohammadi.conveyor.viewmodels
+
+import android.arch.lifecycle.ViewModel
+import android.arch.lifecycle.ViewModelProvider
+
+/**
+ * singleton of view model factory to create instances of view models.
+ */
+object ViewModelFactory: ViewModelProvider.Factory {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt:9]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/9a67d17b3b4351e9a1c31b5b5c4883ada984b461/conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt#L9"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

**Comment by ReviewerOfCode** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt#L4">conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt#L4</a> - Unused import</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/hodamohammadi/Conveyor/blob/8d294783ac09e5f612ce769d1a6a52228b017aba/conveyor/src/main/java/com/hodamohammadi/conveyor/services/ConveyorNotificationService.kt#L5">conveyor/src/main/java/com/hodamohammadi/conveyor/services/ConveyorNotificationService.kt#L5</a> - Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

### AndroidLint found issues

#### Warning (2)

| File | Line | Reason |
| ---- | ---- | ------ |
`conveyor/build.gradle` | 12 | Not targeting the latest versions of Android; compatibility modes apply. Consider testing and updating this version. Consult the android.os.Build.VERSION_CODES javadoc for details. 
`conveyor/src/main/AndroidManifest.xml` | 6 | On SDK version 23 and up, your app data will be automatically backed up, and restored on app install. Your GCM regid will not work across restores, so you must ensure that it is excluded from the back-up set. Use the attribute `android:fullBackupContent` to specify an `@xml` resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ReviewerOfCode** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/hodamohammadi/Conveyor/blob/9a67d17b3b4351e9a1c31b5b5c4883ada984b461/conveyor/src/main/java/com/hodamohammadi/conveyor/services/ConveyorNotificationService.kt#L5">conveyor/src/main/java/com/hodamohammadi/conveyor/services/ConveyorNotificationService.kt#L5</a> - Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

### AndroidLint found issues

#### Warning (2)

| File | Line | Reason |
| ---- | ---- | ------ |
`conveyor/build.gradle` | 12 | Not targeting the latest versions of Android; compatibility modes apply. Consider testing and updating this version. Consult the android.os.Build.VERSION_CODES javadoc for details. 
`conveyor/src/main/AndroidManifest.xml` | 6 | On SDK version 23 and up, your app data will be automatically backed up, and restored on app install. Your GCM regid will not work across restores, so you must ensure that it is excluded from the back-up set. Use the attribute `android:fullBackupContent` to specify an `@xml` resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultDialog.kt [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultDialog.kt:17]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/9a67d17b3b4351e9a1c31b5b5c4883ada984b461/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultDialog.kt#L17"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt [hunk 18]
```diff
@@ -1,18 +1,17 @@
...
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt:15]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/9a67d17b3b4351e9a1c31b5b5c4883ada984b461/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt#L15"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt [hunk 19]
```diff
@@ -7,9 +7,10 @@ import java.io.Serializable
...
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt:14]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/9a67d17b3b4351e9a1c31b5b5c4883ada984b461/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt#L14"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

**Review by itsAlexNguyen — APPROVED** [review]

**Pull request merged**